controllers/ibu_controller.go

```diff
@@ -121,16 +123,105 @@ func (r *ImageBasedUpgradeReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	} else if reconcileTime == utils.StopReconciling {
 		return
 	}
+
+	if imageBasedUpgrade.Spec.Stage == ranv1alpha1.Stages.Idle {
+		return
+	}
+
+	ibus := &ranv1alpha1.ImageBasedUpgradeList{}
```

Comment:
Would it make sense for this CR to be a cluster scoped CR with a fixed name? I think bad things will happen if we ever have more than 1 agent CR.

Reply:
Hmm, that would make it more like a node manager CR (similar to cluster version) instead of a job. There would be more complexity from resetting it in various stages, instead of this little piece of logic.

Reply:
My inclination is that it should still be a cluster scoped CR rather than namespaced (not sure what it would mean for a cluster-wide operation to be defined within a namespace). If we allow multiple versions (ie unique names) then we need tight logic around only 1 is allowed to be in any state other than idle.

Reply:
The logic is already implemented in `isIbuActive`. Is it not tight enough?
We can make it cluster scoped. Right now the operator is single namespaced. That probably needs to change too. In theory, many other operators we've seen should only be installed once only. However most of them are single namespaced with a default namespace that customers shouldn't change, like openshift-ptp ....